### PR TITLE
Update "The Hidden" to make use of Serverside RUI

### PR DESCRIPTION
**Pull request by x3Karma**
Just a small update (hopefully releasing with ~~1.9.3~~ 1.10 ) showing a small snippet of serverside RUI magic being used.

This is not the rework that I'm hopefully planning to do soon.

https://user-images.githubusercontent.com/22678145/181082791-5f116a2e-d444-46ca-bf15-7b7d874d5780.mp4

### CODE REVIEW

**Comment by GeckoEidechse** [comment]
Because the title is a bit vague, this PR only adds the small notification on the top, right? ^^

**Comment by x3Karma** [comment]
> Because the title is a bit vague, this PR only adds the small notification on the top, right? ^^

Ye

**Review by GeckoEidechse — APPROVED** [review]
Notification appears as expected in testing when moving too fast.

(Tested in private match with only one player)

File: Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut [hunk 1]
```diff
@@ -153,35 +159,62 @@ void function RemoveHidden()
...
+					file.isVisible = false
+					player.Signal("VisibleNotification")
 				}
 			}
+			else
+			{
+				player.SetCloakFlicker(0.2 , 1 )
+				player.kv.VisibilityFlags = 0
+				float waittime = RandomFloat(0.5)
+				wait waittime
```
Comment (RoyalBlue1) [inline Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut:193]: when a player disconnects during this wait the server crashes

**Review by RoyalBlue1 — CHANGES_REQUESTED** [review]
This need a bit more formatting

File: Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut (hunk 1 above)
Comment (x3Karma) [reply Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut:195]: Added EndSignals so threads will end if player disconnects

File: Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut [hunk 2]
```diff
@@ -20,8 +24,7 @@ void function GamemodeHidden_Init()
 	AddCallback_GameStateEnter( eGameState.Postmatch, RemoveHidden )
 	SetTimeoutWinnerDecisionFunc( TimeoutCheckSurvivors )
 
-	thread PredatorMain()
-
+	RegisterSignal("VisibleNotification")
```
Comment (GeckoEidechse) [inline Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut:27]:
```suggestion
	RegisterSignal( "VisibleNotification" )
```

File: Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut [hunk 3]
```diff
@@ -78,7 +81,10 @@ void function MakePlayerHidden(entity player)
 
 	SetTeam( player, TEAM_IMC )
 	player.SetPlayerGameStat( PGS_ASSAULT_SCORE, 0 ) // reset kills
+	file.hiddens.append(player)
```
Comment (GeckoEidechse) [inline Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut:84]:
```suggestion
	file.hiddens.append( player )
```

File: Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut [hunk 4]
```diff
@@ -153,35 +159,66 @@ void function RemoveHidden()
...
 	{
 		WaitFrame()
 		if(!IsLobby())
 		{
-			foreach (entity player in GetPlayerArray())
+			if ( !IsValid(player) || !IsAlive(player) || player.GetTeam() != TEAM_IMC )
+				continue
+
+			vector playerVelV = player.GetVelocity()
+			playerVel = sqrt(playerVelV.x * playerVelV.x + playerVelV.y * playerVelV.y + playerVelV.z * playerVelV.z)
```
Comment (GeckoEidechse) [inline Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut:177]:
```suggestion
			playerVel = sqrt( playerVelV.x * playerVelV.x + playerVelV.y * playerVelV.y + playerVelV.z * playerVelV.z )
```

File: Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut [hunk 5]
```diff
@@ -153,35 +159,66 @@ void function RemoveHidden()
...
+
+			if (playerVel/300 < 1.3)
```
Comment (GeckoEidechse) [inline Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut:179]:
```suggestion
			if ( playerVel/300 < 1.3 )
```

File: Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut [hunk 6]
```diff
@@ -153,35 +159,66 @@ void function RemoveHidden()
...
 			{
-				if (player == null || !IsValid(player) || !IsAlive(player) || player.GetTeam() != TEAM_IMC)
-					continue
-				vector playerVelV = player.GetVelocity()
-				float playerVel
-				playerVel = sqrt(playerVelV.x * playerVelV.x + playerVelV.y * playerVelV.y + playerVelV.z * playerVelV.z)
-				float playerVelNormal = playerVel * 0.068544
-				if (playerVel/300 < 1.3)
+				player.SetCloakFlicker(0, 0)
```
Comment (GeckoEidechse) [inline Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut:181]:
```suggestion
				player.SetCloakFlicker( 0, 0 )
```

File: Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut [hunk 7]
```diff
@@ -153,35 +159,66 @@ void function RemoveHidden()
...
+				player.kv.VisibilityFlags = 0
+				wait 0.5
+				if (file.isVisible)
```
Comment (GeckoEidechse) [inline Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut:184]:
```suggestion
				if ( file.isVisible )
```

File: Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut [hunk 8]
```diff
@@ -153,35 +159,66 @@ void function RemoveHidden()
...
-				}
-				else
-				{
-					player.SetCloakFlicker(0.2 , 1 )
-					player.kv.VisibilityFlags = 0
-					float waittime = RandomFloat(0.5)
-					wait waittime
-					player.kv.VisibilityFlags = ENTITY_VISIBLE_TO_EVERYONE
+					file.isVisible = false
+					player.Signal("VisibleNotification")
```
Comment (GeckoEidechse) [inline Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut:187]:
```suggestion
					player.Signal( "VisibleNotification" )
```

File: Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut [hunk 9]
```diff
@@ -153,35 +159,66 @@ void function RemoveHidden()
...
 				}
 			}
+			else
+			{
+				player.SetCloakFlicker(0.2 , 1 )
```
Comment (GeckoEidechse) [inline Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut:192]:
```suggestion
				player.SetCloakFlicker( 0.2 , 1 )
```

File: Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut [hunk 10]
```diff
@@ -153,35 +159,66 @@ void function RemoveHidden()
...
+				player.kv.VisibilityFlags = 0
+				float waittime = RandomFloat(0.5)
```
Comment (GeckoEidechse) [inline Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut:194]:
```suggestion
				float waittime = RandomFloat( 0.5 )
```

File: Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut [hunk 11]
```diff
@@ -153,35 +159,66 @@ void function RemoveHidden()
...
+		{
+			NSDeleteStatusMessageOnPlayer( player, "visibleTitle" )
+			NSDeleteStatusMessageOnPlayer( player, "visibleDesc" )
+			continue
+		}
+		else
+		{
+			NSCreateStatusMessageOnPlayer( player, "You are visible!", "", "visibleTitle")
+			NSCreateStatusMessageOnPlayer( player, "Note:", "Slow down to remain invisible!", "visibleDesc")
+			player.WaitSignal("VisibleNotification")
```
Comment (GeckoEidechse) [inline Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut:220]:
```suggestion
			NSCreateStatusMessageOnPlayer( player, "You are visible!", "", "visibleTitle" )
			NSCreateStatusMessageOnPlayer( player, "Note:", "Slow down to remain invisible!", "visibleDesc" )
			player.WaitSignal( "VisibleNotification" )
```

File: Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut (hunk 1 above)
Comment (GeckoEidechse) [reply Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut:195]: Confirmed fixed with f460f4aceb27cd5767c3e6fdce91434c6a19ba76

**Review by GeckoEidechse — CHANGES_REQUESTED** [review]
Formatting issues

File: Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut [hunk 12]
```diff
@@ -153,35 +159,66 @@ void function RemoveHidden()
...
+        player.EndSignal( "OnDestroy" )
+	float playerVel
+
 	while (true) 
 	{
 		WaitFrame()
 		if(!IsLobby())
 		{
-			foreach (entity player in GetPlayerArray())
+			if ( !IsValid(player) || !IsAlive(player) || player.GetTeam() != TEAM_IMC )
```
Comment (GeckoEidechse) [inline Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hidden.nut:173]:
```suggestion
			if ( !IsValid( player ) || !IsAlive( player ) || player.GetTeam() != TEAM_IMC )
```

**Review by GeckoEidechse — CHANGES_REQUESTED** [review]
Found one more :3

**Review by GeckoEidechse — APPROVED** [review]
Aight worked in testing and formatting is also fixed now ^^

**Pull request merged**